addons/api/addon/serializers/application.js

```diff
@@ -72,6 +73,16 @@ export default class ApplicationSerializer extends RESTSerializer {
       }
       delete json[key];
     }
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excluded
+    // from serialization.
+    if (options?.for && !options.for.includes(recordType)) {
```

Comment:
Can we actually be more specific than using `includes` for strings and do a real equality check? This works fine if `for` is an array but for strings I'm concerned that in the future we can have bugs where we have a type that is a substring of another type.

Reply:
Good point, Done!